### PR TITLE
Add advice on building all modules on Debian-like

### PR DESCRIPTION
**Pull request by Maroloccio**
Current advice regarding installing dependencies is insufficient to build all optional modules:

```
The necessary bits to build these optional modules were not found:                                                            
_bz2                  _curses               _curses_panel
_dbm                  _gdbm                 _hashlib
_lzma                 _sqlite3              _ssl
_tkinter              _uuid                 readline
zlib                                                   
To find the necessary bits, look in setup.py in detect_modules() for the module's name. 
```

Some users may like the convenience of getting a head start on choosing OS packages that get them closer to a complete build.

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept this contribution by verifying everyone involved has signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

## Recognized GitHub username

We couldn't find a [bugs.python.org](https://bugs.python.org/) (b.p.o) account corresponding to the following GitHub usernames:

@Maroloccio

This might be simply due to a missing "GitHub Name" entry in one's b.p.o account settings. This is necessary for legal reasons before we can look at this contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.




You can [check yourself](https://check-python-cla.herokuapp.com/) to see if the CLA has been received.

Thanks again for the contribution, we look forward to reviewing it!

**Comment by Maroloccio** [comment]
@erlend-aasland you :+1: this and the PR has been waiting for a merge for the last couple of months; what can I do to make the last step happen, please?

**Comment by terryjreedy** [comment]
@Mariatta Do you know linux well enough to review this addition to the linux setup part of the devguide?  Or do you know who else commits devguide changes who might? (I am neither.) The idea seem straightforward, and has gotten 4 likes, but from non-committers.

**Review by erlend-aasland — APPROVED** [review]
LGTM, but I'm not sure about the _"versions you prefer"_ bit; CPython has version requirements for some (all?) dependencies. For example SQLite >= 3.7.15.

File: setup.rst
Comment (zware) [inline setup.rst:345]: I think this can be combined with the above point, as I'm pretty sure the explicit package names cover everything that would be installed by `build-dep`.  The above point should also be updated to a more recent version (or the minor version removed entirely, and the suggestion amended to suggest adding a minor version if `python3` doesn't cut it).

Comment (zware) [inline setup.rst:345]: I'm also not a fan of the "versions you prefer..." bit, I would suggest to trim it back to just 
```suggestion
If you want to build all optional modules, install the following packages and
their dependencies::
```

File: setup.rst
Comment (zware) [inline setup.rst:349]: `xvfb` is not actually necessary, it would only allow you to run Tk tests on a headless machine.

**Review by zware — COMMENTED** [review]
Looks like a good addition, but I have some quibbles :)

File: setup.rst
Comment (erlend-aasland) [reply setup.rst:345]: +1

Comment (Maroloccio) [reply setup.rst]: I have removed the minor version from the Python package for which the command installs the build dependencies.

Comment (Maroloccio) [reply setup.rst]: I have accepted this suggestion entirely, merging it with the deletion of the previous paragraph.

File: setup.rst
Comment (Maroloccio) [reply setup.rst]: I have removed `xvfb` from the list.

**Comment by Maroloccio** [comment]
Thank you @zware for your suggestions, please either Resolve the contextual exchanges or recommend further work needed on the PR.

**Review by erlend-aasland — APPROVED** [review]
LGTM with changes applied!

**Review by zware — APPROVED** [review]

**Pull request merged**
